core/src/test/java/eu/fasten/core/data/RevisionCallGraphTest.java

```diff
@@ -1,209 +1,204 @@
-		d = new RevisionCallGraph.Dependency("maven", "foo.bar", ObjectLists.singleton(new RevisionCallGraph.Constraint("[3.1..7.1]")));
-		assertEquals("maven", d.forge);
-		assertEquals("foo.bar", d.product);
-		assertEquals(1, d.constraints.size());
-		assertEquals("3.1", d.constraints.get(0).lowerBound);
-		assertEquals("7.1", d.constraints.get(0).upperBound);
-
-		String spec = "{\"forge\": \"maven\", \"product\": \"foo.bar\", \"constraints\": [\"[3.1..  7.1   ]\",\"[   9]\",\"[10.3  ..]\"] }";
-		JSONObject json = new JSONObject(spec);
-		d = new RevisionCallGraph.Dependency(json);
-		assertEquals("maven", d.forge);
-		assertEquals("foo.bar", d.product);
-		List<RevisionCallGraph.Constraint> constraints = d.constraints;
-		assertEquals(3, constraints.size());
-		assertEquals("3.1", constraints.get(0).lowerBound);
-		assertEquals("7.1", constraints.get(0).upperBound);
-		assertEquals("9", constraints.get(1).lowerBound);
-		assertEquals("9", constraints.get(1).upperBound);
-		assertEquals("10.3", constraints.get(2).lowerBound);
-		assertNull(constraints.get(2).upperBound);
+    @Test
+    public void testConstraint() {
+        RevisionCallGraph.Constraint c;
+
+        c = new RevisionCallGraph.Constraint("3.1", "7.4");
+        assertEquals("3.1", c.lowerBound);
+        assertEquals("7.4", c.upperBound);
+
+        c = new RevisionCallGraph.Constraint("[3.1..7.4]");
+        assertEquals("3.1", c.lowerBound);
+        assertEquals("7.4", c.upperBound);
+
+        c = new RevisionCallGraph.Constraint("[3.1..]");
+        assertEquals("3.1", c.lowerBound);
+        assertNull(c.upperBound);
+
+        c = new RevisionCallGraph.Constraint("[   ..  3.1]");
+        assertNull(c.lowerBound);
+        assertEquals("3.1", c.upperBound);
+        assertEquals("[..3.1]", c.toString());
+
+
+        c = new RevisionCallGraph.Constraint("[3.1]");
+        assertEquals("3.1", c.lowerBound);
+        assertEquals("3.1", c.upperBound);
+
+        Assertions.assertThrows(IllegalArgumentException.class, () ->
+                new RevisionCallGraph.Constraint("joooo"));
+        Assertions.assertThrows(IllegalArgumentException.class, () ->
+                new RevisionCallGraph.Constraint("[a..b..c]"));
+
+        String spec = "[\"[3.1..  7.1   ]\",\"[   9]\",\"[10.3  ..]\"]";
+        JSONArray cs = new JSONArray(spec);
+        List<RevisionCallGraph.Constraint> constraints = RevisionCallGraph.Constraint.constraints(cs);
+        assertEquals(3, constraints.size());
+        assertEquals("3.1", constraints.get(0).lowerBound);
+        assertEquals("7.1", constraints.get(0).upperBound);
+        assertEquals("9", constraints.get(1).lowerBound);
+        assertEquals("9", constraints.get(1).upperBound);
+        assertEquals("10.3", constraints.get(2).lowerBound);
+        assertNull(constraints.get(2).upperBound);
+
+        assertEquals(new JSONArray(spec.replaceAll(" ", "")).toString(), RevisionCallGraph.Constraint.toJSON(constraints).toString());
+    }
```

Comment:
Would it make sense to split this test into multiple smaller tests? All individual blocks seem to be unrelated.

Reply:
Will be taken into account in the RCG refactoring PR